pkg/controller/intrusiondetection/intrusiondetection_controller_test.go

```diff
@@ -274,6 +276,58 @@ var _ = Describe("IntrusionDetection controller tests", func() {
 		})
 	})
 
+	Context("secret availability", func() {
+		It("should not wait on tigera-ee-installer-elasticsearch-access secret when cluster is managed", func() {
+			Expect(c.Create(ctx, &operatorv1.ManagementClusterConnection{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.ManagementClusterConnectionSpec{
+					ManagementClusterAddr: "127.0.0.1:12345",
+				},
+			})).ToNot(HaveOccurred())
+
+			// Delete the secret to ensure that utils.ElasticSearch correctly looks for secrets relevant to managed
+			// clusters only and doesn't attempt to find the IDS job secret.
+			idsSecret := &corev1.Secret{
+				TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: "v1"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      render.ElasticsearchIntrusionDetectionJobUserSecret,
+					Namespace: rmeta.OperatorNamespace(),
+				},
+			}
+			err := c.Delete(ctx, idsSecret)
```

Comment:
I think instead of doing this you should change the before each for the context above this context to not add the installer secret and put in a before each for the sibling contexts that need.

Reply:
Done